six/modules/c++/six.sidd/tests/test_compare_sidd.cpp

```diff
@@ -0,0 +1,208 @@
+
+                // Derived specific classification times
+                six::sidd::DerivedClassification& class1 = 
+                    ddata1->productCreation->classification;
+                six::sidd::DerivedClassification& class2 = 
+                    ddata2->productCreation->classification;
+                class2.createDate = class1.createDate;
+                class2.exemptedSourceDate = class1.exemptedSourceDate;
+                class2.declassDate = class1.declassDate;
+            }
+            else
+            {
+                // TODO: SICD specific time fields
+            }
+        }
+        
+        bool metadataMatches = (*sidd1Metadata) == (*sidd2Metadata);
+
+        if(!metadataMatches)
+        {
+            return false;
+        }
+    }
+    else
+    {
+        if (sidd1Metadata->getNumRows() != sidd2Metadata->getNumRows() ||
+            sidd1Metadata->getNumCols() != sidd2Metadata->getNumCols() ||
+            sidd1Metadata->getNumBytesPerPixel() !=
+                sidd2Metadata->getNumBytesPerPixel())
+        {
+            return false;
+        }
+    }
+
+    int result = 
+        std::memcmp(sidd1Buffer.get(), sidd2Buffer.get(),
+                sidd1Metadata->getNumRows() *
+                sidd1Metadata->getNumCols() *
+                sidd1Metadata->getNumBytesPerPixel());
+
+    return !result;
+
+}
+
+int main(int argc, char* argv[])
+{
+    try 
+    {
+
+        cli::ArgumentParser parser;
+        parser.setDescription("This program compares two SIDDs to determine "
+                "if they are identical or not. If the SIDDs match, the "
+                "program's exit code will be 0. If the SIDDs don't match,"
+                "or if there is an error, the code will be nonzero.");
+        parser.addArgument("--ignore-metadata",
+                "Does not check SIDD metadata for differences",
+                cli::STORE_TRUE, "ignoreMetadata")->setDefault("false");
+        parser.addArgument("--ignore-date",
+                "Does not check SIDD creation date for differences",
+                cli::STORE_TRUE, "ignoreDate")->setDefault("false");
+        parser.addArgument("sidd1", "Input SIDD path", 
+                cli::STORE, "SIDD1", "SIDD1", 1, 1, true);
+        parser.addArgument("sidd2", "Input SIDD path", 
+                cli::STORE, "SIDD2", "SIDD2", 1, 1, true);
+
+        const std::auto_ptr<cli::Results> options(parser.parse(argc, argv));
+
+        const bool ignoreMetadata(options->get<bool>("ignoreMetadata"));
+        const bool ignoreDate(options->get<bool>("ignoreDate"));
+        const std::string sidd1Path(options->get<std::string>("SIDD1"));
+        const std::string sidd2Path(options->get<std::string>("SIDD2"));
+
+        if (siddsMatch(sidd1Path, sidd2Path, ignoreMetadata, ignoreDate))
+        {
+            return 0;
+        }
+        else
+        {
+            return 1;
+        }
+
+
+    }
+    catch (except::Throwable & exception)
```

Comment:
Have the whole bottom of this look like round_trip_six - there are some extra things to catch